examples/indexing_routines.py

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env python
+
+# Copyright 2021 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+import argparse
+import datetime
+import gc
+import math
+
+from benchmark import run_benchmark
+
+try:
+    import cunumeric as np
+except ImportError:
+    import legate.numpy as np
+
+
+def compute_diagonal(steps, N, timing):
+    A1 = np.ones((N,), dtype=int)
+    print("measuring diagonal")
+    start = datetime.datetime.now()
+    for step in range(steps):
+        A2 = np.diag(A1)
```

Comment:
Can you add `warmup` iterations (in all tests), like how `examples/richardson_lucy.py` does it?